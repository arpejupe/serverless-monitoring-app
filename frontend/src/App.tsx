import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { Header, Segment } from 'semantic-ui-react';
import { DataTable, Refresh } from './components';
import './semanticUI';

const title = 'Elections Monitoring App Test';

class App extends Component<any, any> {
  public render() {
    return (
      <Segment>
        <Helmet>
          <title>{title}</title>
        </Helmet>
        <Header textAlign="center" size="huge">
          {title}
        </Header>
        <DataTable />
        <Refresh />
      </Segment>
    );
  }
}

export default App;
