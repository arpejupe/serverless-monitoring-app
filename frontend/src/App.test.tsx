import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'react-testing-library';
import App from './App';

jest.mock('./components', () => {
  return {
    DataTable: () => 'DataTable',
    Refresh: () => 'Refresh',
  };
});

describe('App', () => {
  test('renders without crashing', () => {
    const { getByText } = render(<App />);
    expect(getByText('Elections Monitoring App')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
