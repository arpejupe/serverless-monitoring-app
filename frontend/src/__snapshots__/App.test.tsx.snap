// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="ui segment"
>
  <div
    className="ui huge center aligned header"
  >
    Elections Monitoring App
  </div>
  DataTable
  Refresh
</div>
`;
